packages/core/src/payment-integration/default-payment-integration-service.ts

```diff
@@ -248,4 +250,14 @@ export default class DefaultPaymentIntegrationService implements PaymentIntegrat
 
         return this._storeProjection.getState();
     }
+
+    async stripeLinkAuthenticatedAction(
+        authenticated: boolean | undefined,
+    ): Promise<PaymentIntegrationSelectors> {
+        this._store.dispatch(
+            createAction(CustomerActionType.StripeLinkAuthenticated, authenticated),
```

Comment:
In past this was done in a way since payment integrations were inside core package, can we implement this in such a way that we don't need to have provider specific methods here?

cc @serhii-tkachenko

Reply:
The action which is dispatched by stripeLinkAuthenticatedAction method is used in this place https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/customer/customer-reducer.ts#L48 and then used with combineReducers https://github.com/bigcommerce/checkout-sdk-js/blob/master/packages/core/src/customer/customer-reducer.ts#L30 mehtod which comes from @bigcommerce/data-store’ so I think that this method is neccessary.
But I think that maybe I can make it more generic. Something like this:

    async dispatchAction<TPayload, TMeta, TType extends string>(
        type: TType,
        payload?: TPayload,
        meta?: TMeta,
    ): Promise<PaymentIntegrationSelectors> {
        this._store.dispatch(createAction(type, payload, meta));

        return this._storeProjection.getState();
    }

Reply:
We should not add dispatch action method, because it can cause some inconsistency between payment integrations and core. 

@animesh1987 it is a problem that we talked earlier about payment provider specific state. So in this case we should build extra branch for a state like `getPaymentProviderAuthentication(methodId: string)` and `updatePaymentProviderAuthentication(methodId: string, value: boolean)`. What do you think?

It might take some time on stripe refactoring, but it can resolve the issue that you faced in integration package.

As a reference to add new functionality for a state you can use this PR:
https://github.com/bigcommerce/checkout-sdk-js/pull/2075

But better to wait @animesh1987 to confirm, that this will be the best way to handle this.

Reply:
We don't want to run into a state where we have a umbrella dispatch method, as that can cause a lot of confusion and give access to state where it should not be allowed.

I do agree with @serhii-tkachenko here we can use the state in the PR linked by Serhii or add something similar but not tied to a payment provider.